Common/src/geometry/CPhysicalGeometry.cpp

```diff
@@ -5108,53 +5102,56 @@ unsigned short iMarker, jMarker, iMarkerTP, iSpan, jSpan, kSpan = 0;
-            if (config->GetMarker_All_TurbomachineryFlag(iMarker) == marker_flag){
-              for (iVertex = 0; iVertex < nVertex[iMarker]; iVertex++) {
-                iPoint = vertex[iMarker][iVertex]->GetNode();
-                for (jMarker = 0; jMarker < nMarker; jMarker++){
-                  if (config->GetMarker_All_KindBC(jMarker) == PERIODIC_BOUNDARY) {
-                    PeriodicBoundary = config->GetMarker_All_PerBound(jMarker);
-                    jVertex = nodes->GetVertex(iPoint, jMarker);
-                    if ((jVertex != -1) && (PeriodicBoundary == (val_iZone + 1))){
-                      coord = nodes->GetCoord(iPoint);
-                      switch (config->GetKind_TurboMachinery(val_iZone)){
-                      case CENTRIFUGAL:
-                        valueSpan[nSpan_loc] = coord[2];
-                        break;
-                      case CENTRIPETAL:
-                        valueSpan[nSpan_loc] = coord[2];
-                        break;
-                      case AXIAL:
-                        valueSpan[nSpan_loc] = sqrt(coord[0]*coord[0]+coord[1]*coord[1]);
-                        break;
-                      case CENTRIPETAL_AXIAL:
-                        if (marker_flag == OUTFLOW){
-                          valueSpan[nSpan_loc] = sqrt(coord[0]*coord[0]+coord[1]*coord[1]);
-                        }
-                        else{
-                          valueSpan[nSpan_loc] = coord[2];
-                        }
-                        break;
-                      case AXIAL_CENTRIFUGAL:
-                        if (marker_flag == INFLOW){
-                          valueSpan[nSpan_loc] = sqrt(coord[0]*coord[0]+coord[1]*coord[1]);
-                        }
-                        else{
-                          valueSpan[nSpan_loc] = coord[2];
-                        }
-                        break;
 
-                      }
-                      nSpan_loc++;
-                    }
+          if (config->GetMarker_All_Turbomachinery(iMarker) != iMarkerTP) continue;
+          if (config->GetMarker_All_TurbomachineryFlag(iMarker) != marker_flag) continue;
+
+          for (jMarker = 0; jMarker < nMarker; jMarker++){
+            if (config->GetMarker_All_KindBC(jMarker) != PERIODIC_BOUNDARY) continue;
+
+            for (iVertex = 0; iVertex < nVertex[iMarker]; iVertex++) {
+
+              iPoint = vertex[iMarker][iVertex]->GetNode();
+              if (!nodes->GetDomain(iPoint)) continue;
+
+              PeriodicBoundary = config->GetMarker_All_PerBound(jMarker);
+              jVertex = nodes->GetVertex(iPoint, jMarker);
+
+              if ((jVertex != -1) && (PeriodicBoundary == (val_iZone + 1))){
+                coord = nodes->GetCoord(iPoint);
+                radius = sqrt(coord[0]*coord[0]+coord[1]*coord[1]);
+                switch (config->GetKind_TurboMachinery(val_iZone)){
+                case CENTRIFUGAL:
```

Comment:
`[...] as I will not debug that function any further because it has the world record for nested loops.`
@pcarruscag, 2021-03-04, Digital ink on computerscreen in #1219

Reply:
Continue-man to the rescue 🦸

Reply:
Info: The artist has yet to name his piece 🎨

Reply:
Proper thing to do would be to find the vertices this function wants to then not repeat all those conditions 3 times... But anyway, I scrolled down from this function and realized this was only a drop in the ocean.